timestep.c

```diff
@@ -91,40 +91,12 @@ set_softenings(const double time)
 {
     int i;
 
-    if(All.SofteningGas * time > All.SofteningGasMaxPhys)
-        All.SofteningTable[0] = All.SofteningGasMaxPhys / time;
-    else
-        All.SofteningTable[0] = All.SofteningGas;
-
-    if(All.SofteningHalo * time > All.SofteningHaloMaxPhys)
-        All.SofteningTable[1] = All.SofteningHaloMaxPhys / time;
-    else
-        All.SofteningTable[1] = All.SofteningHalo;
-
-    if(All.SofteningDisk * time > All.SofteningDiskMaxPhys)
-        All.SofteningTable[2] = All.SofteningDiskMaxPhys / time;
-    else
-        All.SofteningTable[2] = All.SofteningDisk;
-
-    if(All.SofteningBulge * time > All.SofteningBulgeMaxPhys)
-        All.SofteningTable[3] = All.SofteningBulgeMaxPhys / time;
-    else
-        All.SofteningTable[3] = All.SofteningBulge;
-
-    if(All.SofteningStars * time > All.SofteningStarsMaxPhys)
-        All.SofteningTable[4] = All.SofteningStarsMaxPhys / time;
-    else
-        All.SofteningTable[4] = All.SofteningStars;
-
-    if(All.SofteningBndry * time > All.SofteningBndryMaxPhys)
-        All.SofteningTable[5] = All.SofteningBndryMaxPhys / time;
-    else
-        All.SofteningTable[5] = All.SofteningBndry;
-
-    for(i = 0; i < 6; i++)
-        All.ForceSoftening[i] = 2.8 * All.SofteningTable[i];
-
-    All.MinGasHsml = All.MinGasHsmlFractional * All.ForceSoftening[0];
+    for(i = 0; i < 6; i ++)
+        All.ForceSoftening[i] = 2.8 * All.GravitySoftening * All.MeanSeparation[1];
+
+    All.ForceSoftening[0] = 2.8 * All.GravitySofteningGas * All.MeanSeparation[1];
+
+    All.MinGasHsml = All.MinGasHsmlFractional * All.ForceSoftening[1];
```

Comment:
Maybe better to do:
All.MinGasHsml = All.MinGasHsmlFractional * All.GravitySoftening * All.MeanSeparation[0]?

Reply:
It depends on the definition; I changed it to be a fraction of DM softening -- so DM Softening sort of acts like the master control block of all 'softening' related parameters.

Reply:
I was worried about odd setups where there are more or less gas particles than DM particles. But since we set this parameter to zero anyway by default I think what you have is fine.